src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -37,17 +37,49 @@ define([
       rollBackApp: React.PropTypes.func.isRequired,
       scaleApp: React.PropTypes.func.isRequired,
       suspendApp: React.PropTypes.func.isRequired,
-      tasksFetchState: React.PropTypes.number.isRequired
+      tasksFetchState: React.PropTypes.number.isRequired,
+      router: React.PropTypes.object.isRequired
     },
 
     getInitialState: function() {
+      var appid = this.props.model.get('id');
+
+      this.tabs = [];
+
+      _.each(tabsTemplate, function(tab) {
+        this.tabs.push({
+          id: tab.id.replace(":appid", appid),
+          text: tab.text
+        });
+      }.bind(this));
+
       return {
         activeViewIndex: 0,
-        activeTabId: tabs[0].id,
+        activeTabId: this.tabs[0].id,
         selectedTasks: {}
       };
     },
 
+    componentWillMount: function() {
+      this.props.router.on("route:app", function (appid, tab) {
+        if(appid) {
+          this.setState({
+            activeTabId: "app/" + appid + (tab ? "/" + tab : "")
+          });
+        }
+      }.bind(this));
+    },
+
+    componentDidMount: function() {
+      this.setState({
+        activeTabId: this.props.router.current()
+      });
+    },
+
+    componentWillUnmount: function() {
+      this.props.router.off("route:app");
```

Comment:
That's defined by me. It returns the current hash fragement. Renamed it to currentHash.